homeassistant/components/hvv_departures/binary_sensor.py

```diff
@@ -0,0 +1,184 @@
+"""Binary sensor platform for hvv_departures."""
+from datetime import timedelta
+import logging
+
+from aiohttp import ClientConnectorError
+import async_timeout
+from pygti.exceptions import InvalidAuth
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_PROBLEM,
+    BinarySensorEntity,
+)
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import ATTRIBUTION, CONF_STATION, DOMAIN, MANUFACTURER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the binary_sensor platform."""
+    hub = hass.data[DOMAIN][entry.entry_id]
+    station_name = entry.data[CONF_STATION]["name"]
+    station = entry.data[CONF_STATION]
+
+    def get_elevator_entities_from_station_information(
+        station_name, station_information
+    ):
+        """Convert station information into a list of elevators."""
+        elevators = {}
+
+        if station_information is None:
+            return {}
+
+        for partial_station in station_information.get("partialStations", []):
+            for elevator in partial_station.get("elevators", []):
+
+                state = elevator.get("state") != "READY"
+                available = elevator.get("state") != "UNKNOWN"
+                label = elevator.get("label")
+                description = elevator.get("description")
+
+                name = f"Elevator {label} at {station_name} ({description})"
```

Comment:
There can be a None in place for label and description. We should probably handle that.